packages/cozy-harvest-lib/src/components/hooks/useMaintenanceStatus.jsx

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react'
+import { Registry } from 'cozy-client'
+import get from 'lodash/get'
+
+const useMaintenanceStatus = (client, slug) => {
+  const [maintenanceStatus, setMaintenanceStatus] = useState({
+    isInMaintenance: false,
+    messages: {}
+  })
+  const registry = new Registry({
+    client
+  })
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const appStatus = await registry.fetchApp(slug)
+      const isInMaintenance = get(appStatus, 'maintenance_activated', false)
+      const messages = get(appStatus, 'maintenance_options.messages', {})
+      setMaintenanceStatus({ isInMaintenance, messages })
```

Comment:
Nit: I think the hook should be in charge just of the fetching and let the data extraction (the two gets) to helpers outside the hook. This would enable those helpers (getMessages and isActive) to be put directly on the Registry class as static methods. This would also remove the need to have a new API (`isMaintenance` and `messages` to communicate with the components.

 I think this would make the hook more flexible as we would not need to have another hook if we want to access other parts of the status response.

Reply:
Yes, agree. There isn't much else in the maintenance status for us so I also think it can stay as it is for now, but it would be a good improvement in the future.